File: app/src/main/java/guepardoapps/bmicalculator/common/controller/DialogController.java
package guepardoapps.bmicalculator.common.controller;

import android.app.Dialog;
import android.content.Context;
import android.support.annotation.NonNull;
import android.view.Window;
import android.widget.Button;
import android.widget.TextView;

import guepardoapps.bmicalculator.R;
import guepardoapps.bmicalculator.common.tools.Logger;

public class DialogController {
    private static final String TAG = DialogController.class.getSimpleName();
    protected Logger _logger;

    private Context _context;
    private Dialog _dialog;

    public Runnable CloseDialogCallback = new Runnable() {
        @Override
        public void run() {
            if (_dialog != null) {
                _dialog.dismiss();
                _dialog = null;
            }
        }
    };

    public DialogController(@NonNull Context context) {
        _logger = new Logger(TAG);
        _logger.Debug("Created new " + TAG + "...");
        _context = context;
    }

    public void ShowDialogDouble(
            @NonNull String title,
            @NonNull String prompt,
            @NonNull String ok,
            @NonNull final Runnable okCallback,
            @NonNull String cancel,
            @NonNull final Runnable cancelCallback,
            boolean isCancelable) {
        _dialog = new Dialog(_context);

        _dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
        _dialog.setContentView(R.layout.dialog_simple);

        TextView titleView = _dialog.findViewById(R.id.custom_alert_dialog_title);
        titleView.setText(title);

        TextView promptView = _dialog.findViewById(R.id.custom_alert_dialog_prompt);
        promptView.setText(prompt);

        Button btnOk = _dialog.findViewById(R.id.custom_alert_dialog_btn_ok);
        btnOk.setText(ok);
        btnOk.setOnClickListener(view -> okCallback.run());

        Button btnCancel = _dialog.findViewById(R.id.custom_alert_dialog_btn_cancel);
        btnCancel.setText(cancel);
        btnCancel.setOnClickListener(view -> cancelCallback.run());

        _dialog.setCancelable(isCancelable);

        _dialog.show();
    }
}

File: app/src/main/java/guepardoapps/bmicalculator/controller/DialogController.kt
package guepardoapps.bmicalculator.controller

import android.app.Dialog
import android.content.Context
import android.view.Window
import android.widget.Button
import android.widget.TextView
import guepardoapps.bmicalculator.R

internal class DialogController(private val context: Context) : IDialogController {
    override fun showDialogDouble(title: String, prompt: String, ok: String, cancel: String, isCancelable: Boolean, okCallback: () -> Unit, cancelCallback: (() -> Unit)?) {
        val dialog = Dialog(context)

        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
        dialog.setContentView(R.layout.dialog_simple)

        val titleView = dialog.findViewById<TextView>(R.id.custom_alert_dialog_title)
        titleView.text = title

        val promptView = dialog.findViewById<TextView>(R.id.custom_alert_dialog_prompt)
        promptView.text = prompt

        val btnOk = dialog.findViewById<Button>(R.id.custom_alert_dialog_btn_ok)
        btnOk.text = ok
        btnOk.setOnClickListener {
            okCallback.invoke()
            dialog.dismiss()
        }

        val btnCancel = dialog.findViewById<Button>(R.id.custom_alert_dialog_btn_cancel)
        btnCancel.text = cancel
        btnCancel.setOnClickListener {
            cancelCallback?.invoke()
            dialog.dismiss()
        }

        dialog.setCancelable(isCancelable)

        dialog.show()
    }
}